erpnext/erpnext_integrations/doctype/authorizenet_settings/authorizenet_settings.py

```diff
@@ -0,0 +1,216 @@
+
+"""
+from __future__ import unicode_literals
+import frappe
+import imp
+import json
+import os
+import sys
+import re
+from six.moves.urllib.parse import urlencode
+from frappe.model.document import Document
+from frappe.integrations.utils import create_payment_gateway, create_request_log
+from frappe.utils import get_url, call_hook_method
+from frappe.utils.password import get_decrypted_password
+
+from authorizenet import apicontractsv1
+from authorizenet.apicontrollers import createTransactionController
+
+class AuthorizenetSettings(Document):
+	supported_currencies = ["USD","CAD"]
+
+	def validate_transaction_currency(self, currency):
+		if currency not in self.supported_currencies:
+			frappe.throw(_("Please select another payment method. AuthorizeNet does not support transactions in currency '{0}'").format(currency))
+	
+	def validate(self):
+		create_payment_gateway('Authorizenet')
+		call_hook_method('payment_gateway_enabled', gateway="Authorizenet")
+
+	def get_payment_url(self, **kwargs):
+		return get_url("./integrations/authorizenet_checkout?{0}".format(urlencode(kwargs)))
+
+
+@frappe.whitelist()
+def charge_credit_card(data, card_number, expiration_date, card_code):
+	"""
+	Charge a credit card
+	"""
+	data = json.loads(data)
+	data = frappe._dict(data)
+
+	# Create Integration Request
+	integration_request = create_request_log(data, "Host", "AuthorizeNet")
+
+	# Authenticate with Authorizenet
+	merchant_auth = apicontractsv1.merchantAuthenticationType()
+	merchant_auth.name = frappe.db.get_single_value("Authorizenet Settings", "api_login_id")
+	merchant_auth.transactionKey = get_decrypted_password('Authorizenet Settings', 'Authorizenet Settings', fieldname='api_transaction_key', raise_exception=False)
+	
+	# Create the payment data for a credit card
+	credit_card = apicontractsv1.creditCardType()
+	credit_card.cardNumber = card_number
+	credit_card.expirationDate = expiration_date
+	credit_card.cardCode = card_code
+
+	# Add the payment data to a paymentType object
+	payment = apicontractsv1.paymentType()
+	payment.creditCard = credit_card
+
+	pr = frappe.get_doc("Payment Request", data.reference_docname)
+	reference_doc = frappe.get_doc(pr.reference_doctype, pr.reference_name).as_dict()
+
+	customer_address = apicontractsv1.customerAddressType()
+	customer_address.firstName = data.payer_name
+	customer_address.email = data.payer_email
+	customer_address.address = reference_doc.customer_address[:60]
+
+	# Create order information 
+	order = apicontractsv1.orderType()
+	order.invoiceNumber = reference_doc.name
+
+	# build the array of line items
+	line_items = apicontractsv1.ArrayOfLineItem()
+
+	for item in reference_doc.get("items"):
+		for i in range(len(reference_doc.get("items"))):
+
+			# setup individual line items
+			item[i] = apicontractsv1.lineItemType()
+			item[i].itemId = item.item_code
+			item[i].name = item.item_name[:30]
+			item[i].description = item.item_name
```

Comment:
Confirm if there's a limit on this key, cause descriptions can get long

```suggestion
			item[i].description = item.description
```